TCC.Lib/Process/ProcessStartHelper.cs

```diff
@@ -0,0 +1,169 @@
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    CreateNoWindow = true
+                },
+                EnableRaisingEvents = true
+            };
+
+            var standardOutput = new List<string>();
+            var standardError = new List<string>();
+            var standardInfos = new List<string>();
+            var standardOutputResults = new TaskCompletionSource<List<string>>();
+            var standardErrorResults = new TaskCompletionSource<List<string>>();
+
+            process.OutputDataReceived += (sender, e) =>
+            {
+                if (e.Data == null)
+                {
+                    standardOutputResults.SetResult(standardOutput);
+                }
+                else
+                {
+                    standardOutput.Add(e.Data);
+                }
+            };
+
+            process.ErrorDataReceived += (sender, e) =>
+            {
+                if (e.Data == null)
+                {
+                    standardErrorResults.SetResult(standardError);
+                }
+                else
+                {
+                    if (outputClassifier?.IsIgnorable(e.Data) == true)
+                    {
+                        return;
+                    }
+                    if (outputClassifier?.IsInfo(e.Data) == true)
+                    {
+                        standardInfos.Add(e.Data);
+                    }
+                    else
+                    {
+                        standardError.Add(e.Data);
+                    }
+                }
+            };
+
+            var tcs = new TaskCompletionSource<CommandResult>();
+
+            async void OnProcessExit(object sender, EventArgs args)
+            {
+                var stdout = await standardOutputResults.Task;
+                var stderr = await standardErrorResults.Task;
+
+                var result = new CommandResult
+                {
+                    Command = command,
+                    ExitCode = process.ExitCode,
+                    IsSuccess = standardError.Count == 0,
+                    Output = string.Join(Environment.NewLine, stdout),
+                    Errors = string.Join(Environment.NewLine, stderr),
+                    Infos = standardInfos,
+                };
+
+                tcs.TrySetResult(result);
+                process.Exited -= OnProcessExit;
+            }
+
+            process.Exited += OnProcessExit;
+
+            using (cancellationToken.Register(() =>
+            {
+                tcs.TrySetCanceled();
+                process.Exited -= OnProcessExit;
+                try
+                {
+                    KillProcessAndChildren(process.Id);
+
+                    if (!process.HasExited)
+                    {
+                        process.Kill();
+                    }
+                }
+                catch (InvalidOperationException)
+                {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS108)